cache/remotecache/local/local.go

```diff
@@ -0,0 +1,73 @@
+package local
+
+import (
+	"context"
+	"time"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/reference"
+	"github.com/moby/buildkit/cache/remotecache"
+	"github.com/moby/buildkit/session"
+	sessioncontent "github.com/moby/buildkit/session/content"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+// ResolveCacheExporterFunc for "local" cache exporter.
+// ref is ignored at the moment
```

Comment:
is there a plan for the `ref` to not be ignored at some point, i.e. is the unfortunate need for `--export-cache ignoredtag` supposed to go away soon?

Reply:
Yes, I can implement support for `index.json`, but two issues:
* needs to define convention for the name of lock file. maybe just `index.json.lock`
* how to export without tag?

Reply:
My first though was that rather than `--export-cache tag` we could make `--export-cache` a bool with no argument and then require `--export-cache-opt tag=...` for the push to repo case, in effect making `tag` optional for modes which don't need it.

Reply:
Yes, it should be the easiest if we can break CLI.

@tonistiigi WDYT?

Reply:
I'm fine with breaking `buildctl` flags. But if we have `--export-cache-type`, shouldn't we switch to `--export-cache-ref` and `--export-cache-mode`? The other way could be `--export-cache local:/path/to/cache` or just value with `^(\.|/)` is used as a local path(although we may want to add more transports in the future).